### PR TITLE
Bug 1307493: Apply exponential back off to sidecar startup. r=garndt

### PR DESCRIPTION
**Pull request by walac**
sidecar containers are the containers that start together with the task
container and that implement the task features. The service implemented
inside the sidecar cotainer may fail due to network hiccups, causing the
sidecar containers to fail to startup.

By applying startup retry with exponential back off we hope to fix these
random startup fails. Given the low probability of an startup fail, we
retry only 3 times.

### CODE REVIEW

File: src/lib/task.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ import fs from "mz/fs";
 import child_process from "mz/child_process";
 import taskcluster from 'taskcluster-client';
 import base from 'taskcluster-base';
+import promiseRetry from 'promise-retry';
```
Comment (gregarndt) [inline src/lib/task.js]: Wow, I didn't know about this library, but that certainly makes things easier!  @imbstack have you ever dealt with this library before? I'm not aware of the nuances (if any).

File: src/lib/task.js [hunk 2]
```diff
@@ -816,7 +817,13 @@ export class Task extends EventEmitter {
       // Build the list of container links... and base environment variables
       linkInfo = await this.states.link(this);
       // Hooks prior to running the task.
-      await this.states.created(this);
+      await promiseRetry(retry => {
+        return this.states.created(this).catch(retry);
```
Comment (gregarndt) [inline src/lib/task.js]: Do all of our state handlers implement a created() method that can be retried?  we should audit our features to ensure that this won't introduce some other problems.

File: src/lib/task.js (hunk 1 above)
Comment (walac) [reply src/lib/task.js]: It is used in artifact upload retries.

Comment (imbstack) [reply src/lib/task.js]: Interesting! And nope, I've never used it before.

**Pull request merged**